src/main/groovy/com/rundeck/plugins/ansible/ansible/AnsibleRunner.java

```diff
@@ -659,13 +668,27 @@ public boolean registerKeySshAgent(String keyPath) throws AnsibleException, Exce
         }
       }
 
+      Thread errthread = Logging.copyStreamThread(proc.getErrorStream(), ListenerFactory.getListener(System.err));
+      Thread outthread = Logging.copyStreamThread(proc.getInputStream(), ListenerFactory.getListener(System.out));
+      errthread.start();
+      outthread.start();
+
       int exitCode = proc.waitFor();
 
+      outthread.join();
+      errthread.join();
+      System.err.flush();
+      System.out.flush();
+
      if (exitCode != 0) {
           throw new AnsibleException("ERROR: ssh-add returns with non zero code:" + procArgs.toString(),
                   AnsibleException.AnsibleFailureReason.AnsibleNonZero);
       }
 
+      if(tempPassVarsFile!=null && tempPassVarsFile.exists()){
```

Comment:
Should this be above the previous if condition? I believe if the user provides an invalid password, ssh-add will return `1` which will cause it to throw and leave the temp file kicking around.